app/Models/Port.php

```diff
@@ -209,13 +209,25 @@ public function scopeHasAccess($query, User $user)
         return $this->hasPortAccess($query, $user);
     }
 
+    public function scopeInDeviceGroup($query, $deviceGroup)
```

Comment:
Port is already a DeviceRelatedModel, so you shouldn't need this to be redefined here.